core/imt/src/TThreadExecutor.cxx

```diff
@@ -94,23 +120,19 @@ namespace ROOT {
 
    void TThreadExecutor::ParallelFor(unsigned int start, unsigned int end, unsigned step, const std::function<void(unsigned int i)> &f)
    {
-      tbb::parallel_for(start, end, step, f);
+      tbb::this_task_arena::isolate([&]{
+         tbb::parallel_for(start, end, step, f);
+      });
    }
 
    double TThreadExecutor::ParallelReduce(const std::vector<double> &objs, const std::function<double(double a, double b)> &redfunc)
    {
-      return tbb::parallel_reduce(tbb::blocked_range<decltype(objs.begin())>(objs.begin(), objs.end()), double{},
-      [redfunc](tbb::blocked_range<decltype(objs.begin())> const & range, double init) {
-         return std::accumulate(range.begin(), range.end(), init, redfunc);
-      }, redfunc);
+      return ROOT::Internal::ReduceHelper<double>(objs, redfunc);
```

Comment:
I'm ok with the change but, although I agree that maybe it can be decluttered a little, I slightly prefer the previous version of the code because you could inmediately see that what you were returning was a tbb::parallel_reduce instead of adding another level of indirection (The 3rd one: Reduce->ParallelReduce->ReduceHelper->tbb::parallel_reduce).

In any case I would change the name of ReduceHelper to ParallelReduceHelper (as parallel reduce is the special case and not the normal reduction operation applied)